### PR TITLE
docs: add snapshot-related info sections to the Pull Requests documentation

### LINKED ISSUE #8997: Docs: contributing docs should say how to update rule doc code example snapshots
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Since https://github.com/typescript-eslint/typescript-eslint/issues/8382, we have snapshots of the eslint output on the code examples in the rule docs. However, our documentation doesn't say how to update those snapshots; if you change the code examples, the tests just fail locally and in CI. 

I've been using `yarn test -- -u` to get them to update, but I'm not actually sure if that's the recommended way. (or should they just automatically update when running `yarn test` like our other snapshots do, in which case we should just make a code change?)

Ping @auvred 

### Affected URL(s)

https://typescript-eslint.io/contributing/local-development#validating-changes

### PR DESCRIPTION
**Pull request by alythobani**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8997
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

- Added "AST Testing" and "Updating Snapshots" sections to the [Pull Requests doc](https://typescript-eslint.io/contributing/pull-requests)
- Fixed heading levels

🌱

### CODE REVIEW

File: docs/contributing/Pull_Requests.mdx [hunk 1]
```diff
@@ -44,7 +44,21 @@ For rule tests we recommend, when reasonable:
...
 
-### Raising the PR
+### AST Testing
+
+AST tests belong in the `ast-spec` package, within `fixtures/` subfolders alongside the AST declarations. Each test has its own `<test-name>/` folder, containing a `fixture.ts` file defining the test, and a `snapshots/` subfolder. Happy-path test folders are stored directly under `fixtures/`; error-path test folders go under `fixtures/_errors/`. You can check out the [`ClassDeclaration` fixtures folder](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/ast-spec/src/declaration/ClassDeclaration/fixtures) for an example of both happy-path and unhappy-path test folders.
+
+### Updating Snapshots
+
+Jest snapshots are generated for use in some tests, e.g. for [rule schemas](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin/tests/schema-snapshots) and [code examples in rule docs](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin/tests/docs-eslint-output-snapshots). You may need to re-generate these snapshots after adjusting a rule and/or its documentation, by running the relevant test suite(s) with the `-u` flag:
```
Comment (JoshuaKGoldberg) [inline docs/contributing/Pull_Requests.mdx:53]: > `.../main/...`

[Docs] It's best to use commit permalinks, as file intent and/or paths can change over time. E.g. `https://github.com/typescript-eslint/typescript-eslint/tree/04990d545fc119329551ae3a55d79dfc0c7bf147/packages/eslint-plugin/tests/schema-snapshots`.

Comment (JoshuaKGoldberg) [inline docs/contributing/Pull_Requests.mdx:53]: > Jest snapshots

[Docs] How about linking to the Jest docs to make it clear we're using a first-class feature? https://jestjs.io/docs/snapshot-testing

File: docs/contributing/Pull_Requests.mdx [hunk 2]
```diff
@@ -44,7 +44,21 @@ For rule tests we recommend, when reasonable:
...
+
+```bash
+cd packages/eslint-plugin
+yarn test docs -u
+yarn test schemas -u
+```
+
+## Raising the PR
```
Comment (JoshuaKGoldberg) [inline docs/contributing/Pull_Requests.mdx:61]: [Question] Before it was _Raising **a** PR_; here it's _Raising **the* PR_. Any reason for the change?

I'm not for or against it in theory, just curious 😄. I do like the increase in header level.

File: docs/contributing/Pull_Requests.mdx
Comment (JoshuaKGoldberg) [inline docs/contributing/Pull_Requests.mdx:1]: [Praise] Thanks for the 🌱 emoji! It's a great one. 🙂

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
A great start, thanks! I like the fixing up of header levels and the general explanations. Clear and informative. Nicely done! 🙌

File: docs/contributing/Pull_Requests.mdx (hunk 1 above)
Comment (Josh-Cena) [reply docs/contributing/Pull_Requests.mdx:53]: @JoshuaKGoldberg I slightly prefer linking to `main` in this case:

- If the file is missing, we _want_ to update the docs
- If we updated the code, we don't want anyone to read outdated code.

Comment (JoshuaKGoldberg) [reply docs/contributing/Pull_Requests.mdx:53]: Hmm, I guess that makes sense. I feel more comfortable about these kinds of permalinks when Docusaurus flags on missing links...

File: docs/contributing/Pull_Requests.mdx
Comment (alythobani) [reply docs/contributing/Pull_Requests.mdx:1]: one of my favourites hehe

File: docs/contributing/Pull_Requests.mdx (hunk 2 above)
Comment (alythobani) [reply docs/contributing/Pull_Requests.mdx:61]: I think that may have been from your own [PR](https://github.com/typescript-eslint/typescript-eslint/pull/9246/files#diff-53ca1501da8503e35977ab9be0faaa5c83ff62855456aa6e9ab089374db91f2aL28-R47) a couple days ago 😃  no strong feelings myself either!

File: docs/contributing/Pull_Requests.mdx (hunk 1 above)
Comment (alythobani) [reply docs/contributing/Pull_Requests.mdx]: Great call! Done

Comment (alythobani) [reply docs/contributing/Pull_Requests.mdx]: Happy to do whatever we think is best here!

File: docs/contributing/Pull_Requests.mdx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply docs/contributing/Pull_Requests.mdx:61]: Lol. Well. Now I don't remember.

**Review by JoshuaKGoldberg — APPROVED** [review]
✨ looks great to me, thanks!

**Comment by JoshuaKGoldberg** [comment]
I say, let's just ship 😄. If the repo ever decides on a standardized approach we'd lint for it.

**Pull request merged**